src/atlas-orca/meshgenerator/OrcaMeshGenerator.cc

```diff
@@ -241,9 +241,11 @@ void OrcaMeshGenerator::generate( const Grid& grid, const grid::Distribution& di
 
                     // grid ij coordinates
                     {
-                      const auto ij_glb = local_orca.orca_haloed_global_grid_ij( ix, iy );
-                      nodes.ij( inode, XX ) = ij_glb.i;
-                      nodes.ij( inode, YY ) = ij_glb.j;
+                      //   const auto ij_glb = local_orca.orca_haloed_global_grid_ij( ix, iy );
+                      //   nodes.ij( inode, XX ) = ij_glb.i;
+                      //   nodes.ij( inode, YY ) = ij_glb.j;
+                      nodes.ij( inode, XX ) = ix + local_orca.ix_min();
+                      nodes.ij( inode, YY ) = iy + local_orca.iy_min();
```

Comment:
Is it possible to express the difference in this change in terms of some sort of named coordinate? Sorry my head always start to spin as soon as I look at this mesh generator again - I was hoping I had captured all the cases in terms of functions (although not efficient we were going for clarity first because we had so many issues getting it correct for higher res grids). 

`orca_haloed_global_grid_ij` is supposed to wrap points if they extend outside the original orca halo in the case of mpi with halo > 0, but behave as you have it here when inside that region.

It could be that the case I was catching no longer happens now that the code is setup as I have it, but in that case I think what you want is `global_ij` because you are not concerned about this wrapping issue?

Reply:
I don't think the proposed change will work if halo size > 0. If halo_size > 0,  ix+ local_orca.ix_min() or iy + local_orca.iy_min() can be a point that is outside the bounds of the ORCA grid. By using local_orca.orca_haloed_global_grid_ij, any points that lie outside the ORCA grid bounds get wrapped to the correct points on the ORCA grid.

Reply:
The i,j coordinates are not needing to be wrapped. They are just there to see the grid structure.

To see the problem, we can look at the following command:
```bash
mpirun -np 2 bin/atlas-meshgen ORCA2_T --coordinates=ij --info --ghost --halo=1
gmsh mesh.msh mesh_info.msh &
```
and e.g. look again at the 'water' field.

With my modification [6509f04](https://github.com/ecmwf/atlas-orca/pull/20/commits/6509f048ad54c55e3e91dd81a986d827200fb7d5) applied:
![image](https://github.com/user-attachments/assets/f13d22e9-1b5a-4874-82ce-2c79f3dc9fd5)
Without the modification:
![image](https://github.com/user-attachments/assets/c4a5642c-743a-4138-85fd-f6503adf1216)

As you can see, without the modification there's no easy way to see the ghost region. With the modification and having i,j counting further into the ghost region it is easier to see what's going on.

Still it seems there are some issues in the corners even with the modification applied...

Reply:
> I don't think the proposed change will work if halo size > 0. If halo_size > 0, ix+ local_orca.ix_min() or iy + local_orca.iy_min() can be a point that is outside the bounds of the ORCA grid. By using local_orca.orca_haloed_global_grid_ij, any points that lie outside the ORCA grid bounds get wrapped to the correct points on the ORCA grid.

It is OK to go outside the bounds of the original ORCA grid (up to any desired halo size) as long as it gets correctly marked as a periodic halo, and the halo-exchanges are done correctly. In the end what gets written to file is the grid without halo (or just the halos present from in the grid as was originally defined in netCDF format)

Reply:
> Still it seems there are some issues in the corners even with the modification applied...

This might be from my fixes :). In order to get those boxes to look all neat, you end up having ghost nodes that point to other ghost nodes in a previous implementation. This is very bad, because a lot of atlas is dereferencing ghost nodes one time to get to the master node in various places.

Reply:
> > I don't think the proposed change will work if halo size > 0. If halo_size > 0, ix+ local_orca.ix_min() or iy + local_orca.iy_min() can be a point that is outside the bounds of the ORCA grid. By using local_orca.orca_haloed_global_grid_ij, any points that lie outside the ORCA grid bounds get wrapped to the correct points on the ORCA grid.
> 
> It is OK to go outside the bounds of the original ORCA grid (up to any desired halo size) as long as it gets correctly marked as a periodic halo, and the halo-exchanges are done correctly. In the end what gets written to file is the grid without halo (or just the halos present from in the grid as was originally defined in netCDF format)

Thanks for this explanation and for fixing the code.